tests/legacy-cli/e2e/utils/project.ts

```diff
@@ -165,9 +165,11 @@ export function useCIDefaults(projectName = 'test-project') {
     const appTargets = project.targets || project.architect;
     appTargets.build.options.progress = false;
     appTargets.test.options.progress = false;
-    // Disable auto-updating webdriver in e2e.
     if (appTargets.e2e) {
+      // Disable auto-updating webdriver in e2e.
       appTargets.e2e.options.webdriverUpdate = false;
+      // Use a random port in e2e.
+      appTargets.e2e.options.port = 0;
```

Comment:
@alan-agius4 do you think this would also be the ideal place to set `appTargets.serve.options.port`? I just had a test running `ng serve` fail in another PR due to 4200 being in use...

Reply:
Note there's various `ng serve --port=0` throughout the tests already but a few are missing such as
```
rebuild-deps-type-check.ts
rebuild-types.ts
```

Reply:
I've added another commit but feel free to only merge the first if you'd like.

Reply:
Yes we should.